File: server/src/Tables/Projectile.cs
using SpacetimeDB;

namespace masks.server.Tables;

[Table(Public = true)]
[SpacetimeDB.Index.BTree(Name = "PlayerId", Columns = [nameof(PlayerId)])]
public partial struct Projectile
{
    [PrimaryKey]
    public uint EntityId;
    public uint PlayerId;
    public DbVector2 Direction;
    public DbVector2 Position;
    public float Speed;
}

File: server/src/tables/projectile.rs
use crate::math::dbvector2::DbVector2;
#[spacetimedb::table(name = projectile, public)]
pub struct Projectile {
    #[primary_key]
    pub entity_id: u32,
    #[index(btree)]
    pub player_id: u32,
    pub direction: DbVector2,
    pub position: DbVector2,
    pub speed: f32,
}
